guides/extensions/embedded-scripts/embedded-script-extension-files-and-code.md

```diff
@@ -0,0 +1,149 @@
+
+```tsx
+npm run generate
+```
+
+## embedded.html
+
+The file named `embedded.html` contains the code you wish to inject. Your code will be added to the page's head or body depending on your configuration.
+
+Your file structure should look like this:
+
+In your HTML code, you can:
+- [Reference local files](#referencing-local-files-in-your-html-code)
+- [Use dynamic parameters](#using-dynamic-parameters-in-your-html-code)
+- [Add global CSS](#add-global-css-to-your-html-code)
+
+### Referencing local files in your HTML code
+
+Wix will host and deploy every file in your project unless you specify otherwise, including any that you add. Your HTML code can reference these files using a relative path. Your HTML code can reference these files using a relative path.
+
+When referencing local files in a `<script>` tag, the tag needs to have `type="module"`.
+
+For example, to reference a file named `local-script.js` in the same directory as `embedded.html`, use the following code:
+
+  ```tsx
+  <script type = "module" src = "./local-script.js"></script>
+  ```
+>**Note:** TypeScript files are supported.
+
+### Using dynamic parameters in your HTML code
+
+You can embed dynamic parameters in your code to inject custom information per site.
+
+For example:
+
+  ```tsx
+  <meta name = "google-tag" id = "{{googleTag}}"></meta>
+  <script> console.log("Hello {{userName}} from the CLI.");</script>
+  ```
+
+Dynamic parameters must be:
+
+* Wrapped in double curly braces. For example, `{{userName}}`.
+* Within a quote (") to avoid code evaluation.
+* Set using the [Embedded Scripts API](https://dev.wix.com/api/rest/app-management/apps/embedded-scripts/introduction). This is explained in detail in step 5.
+* Made up only of letters. (No special characters or spaces.)
+
+Dynamic parameter values must be strings.
+
+See [below](#paramsdevjson) how to specify dynamic parameter values to use during development
+
+### Adding global CSS to your HTML code
+
+You can add CSS directly to your `embedded.html` file, or you can reference a CSS stylesheet with a link. For example:
+
+  ```tsx
+  <link rel="stylesheet" href="./<your-css-file>.css"/>
+  ```
+
+This CSS applies to your site globally. For example, the following code would make the background of every page of your site red:
+
+  ```tsx
+  #c1dmp {
+    background: red;
+  }
+  ```
+
+## embedded.json
+
+This file contains information about your script, and must have the following structure:
+
+  ```tsx
+  {
+    "id": "1347ffed-668c-4133-9d90-510f58f02c33",
+    "name": "console-logger",
+    "scriptType": "FUNCTIONAL",
+    "placement": "HEAD"
+  }
+  ```
+
+* `id`  is a unique identifier for your script. For example, a randomly generated GUID.
+* `name` is the name of your script as it will appear in the [Wix Developers Center](https://dev.wix.com/apps/my-apps). It can only contain letters and the hyphen (-) character.
```

Comment:
Dev center is huge, I would say "a name will help you identify your embedded script in your dev center extensions page"
![Uploading Screenshot 2024-04-02 at 17.14.43.png…]()